webrender/src/clip_scroll_tree.rs

```diff
@@ -214,16 +184,33 @@ impl ClipScrollTree {
         false
     }
 
-    pub fn scroll(
+    fn find_nearest_scrolling_ancestor(
+        &self,
+        index: Option<ClipScrollNodeIndex>
+    ) -> ClipScrollNodeIndex {
+        let index = match index {
+            Some(index) => index,
+            None => return self.topmost_scroll_node_index(),
+        };
+
+        let node = &self.nodes[index.0];
+        match node.node_type {
+            NodeType::ScrollFrame(state) if state.sensitive_to_input_events() => return index,
+            _ => {}
+        }
+
+        return self.find_nearest_scrolling_ancestor(node.parent)
```

Comment:
nit: let's move this into `_ => ` arm and remove `return` calls

Reply:
Sure thing.